src/tool/xmeta/cpp/ast_to_st_listener.cpp

```diff
@@ -374,61 +404,431 @@ listener_error ast_to_st_listener::extract_property_accessors(XlangParser::Inter
+                    {
+                        char number[256] = ""; /* In fact not necessary as snprintf() adds the
+                                               0-terminator. */
+
+                        snprintf(number, sizeof number, "%zu", create_instance_index);
+                        factory_name = create_instance + number;
+                    }
+                    else
+                    {
+                        factory_name = create_instance;
+                    }
+                    type_ref tr = type_ref{ class_name };
+                    auto syn_constructor_model = std::make_shared<method_model>(factory_name, class_constructor->IDENTIFIER()->getSymbol()->getLine(), xlang_model.m_assembly, std::move(tr), method_sem, method_association::Constructor);
+                    if (class_constructor->formal_parameter_list())
+                    {
+                        extract_formal_params(class_constructor->formal_parameter_list()->fixed_parameter(), syn_constructor_model);
+                    }
+                    synthesized_interface_factory->add_member(syn_constructor_model);
+                    has_factory = true;
+                    create_instance_index++;
+                }
+                else
+                {
+                    auto syn_constructor_model = std::make_shared<method_model>(constructor_semantic_name, class_constructor->IDENTIFIER()->getSymbol()->getLine(), xlang_model.m_assembly, std::move(std::nullopt), method_sem, method_association::Constructor);
+                    if (class_constructor->formal_parameter_list())
+                    {
+                        extract_formal_params(class_constructor->formal_parameter_list()->fixed_parameter(), syn_constructor_model);
+                    }
+                    synthesized_interface->add_member(syn_constructor_model);
+                    has_instance = true;
+                }
+
+                /* TODO: We don't need to update add any members to the class model, the class model can completely be void of methods 
+                properties and events and the metadata emit can user the interfaces that are referenced in the class
+                 to emit metadata */
+                if (clss_model->add_member(constructor_model) == semantic_error::symbol_exists)
+                {
+                    error_manager.write_type_member_exists_error(decl_line, constructor_model->get_name(), clss_model->get_qualified_name());
+                }
+            }
+            if (class_member->class_method_declaration())
+            {
+                auto class_method = class_member->class_method_declaration();
+                std::string method_id = class_method->IDENTIFIER()->getText();
+
+                std::optional<type_ref> tr = type_ref{ class_method->return_type()->getText() };
+                extract_type(class_method->return_type(), tr);
+
+                // TODO: semantic checks
```

Comment:
repeated pattern - factor out a semantic_check() ?

Reply:
This may require a deeper rework of the grammar which i have something in mind that will make this easier (Combining all method, properties, and events into one grammar production and just semantically check what is allowed later). Will do so this in a refactor later.